Zend/tests/036.phpt

```diff
@@ -11,4 +11,4 @@ try {
 
 ?>
 --EXPECT--
-Illegal offset type
+Cannot access offset of type object on array
```

Comment:
I propose `Illegal array offset of type object` to stress the type (as array offset) is not supported.

Reply:
I prefer too.

Reply:
Can we have this as a *follow* up PR as this is the wording used for string offsets:
```
zend_type_error("Cannot access offset of type %s on string", zend_zval_value_name(offset));
```

The whole exercise of this PR is to align array offset and string offset error messages.

Reply:
"Illegal offset type" was always present and this PR this to describe the error newly with "in unset" (and friends), so I belive the "Cannot access offset" should not be introduced in the first place.